Improves sleeper code and makes syndie sleepers deconstructable

JohnFulpWillard opened the pull request:
## About The Pull Request

This PR does four things:
1. Improves sleeper code by a lot, considering last time they were touched was years ago (last edit was 3 years ago when it was replaced with stasis units). This means a removal of ``scrambled_chems`` var in favor of just checking if the machine is emagged.
2. Removes the entirely unused ``sleep_console``.
3. Lowercases the dm file (its been bugging me forever, I didn't want to change it until I did a code improvement PR for it).
4. Lets Syndicate sleepers get deconstructed regardless of mapload.

## Why It's Good For The Game

For the Syndicate sleeper, I have 2 reasons for this:
1. Syndicate agents who accidentally destroy their sleepers can now rebuild it
2. People who successfully hijack the shuttle, as a nice bonus for people who manage to defeat nukies and take over their ship.

I personally see this as QoL but I can see it being labelled balance as well, if anyone wants to change that you can feel free.

## Changelog

:cl:
qol: Syndicate sleepers (the ones in the nukie base outpost) now drops their board on deconstruction.
/:cl:

JohnFulpWillard committed:
Improves sleeper code and makes syndie sleepers deconstructable

* Improves regular Sleeper code by a lot considering last time they were touched was years ago, removes the entirely unused sleep_console, and lets Syndicate sleepers get deconstructed regardless of mapload.
* Also finally lowercases the dm file.

Mothblocks commented:
> Lowercases the dm file (its been bugging me forever, I didn't want to change it until I did a code improvement PR for it).

5 dollars says this causes the same case sensitive filename windows crap we deal with every time this happens

Mothblocks commented:
oh wow look at that it already is. in your pr

Mothblocks commented:
i think you should just not do the rename at all unless you pick a different filename like `sleepers.dm`

JohnFulpWillard committed:
watch this swag

JohnFulpWillard committed:
no clue if this will work but I'm gonna try before renaming

JohnFulpWillard commented:
outsmarted the CI

Mothblocks commented:
it's not the ci dude, it's *us* when we switch branches and git throws a fit on windows

newfren commented:
there are syndicate sleepers in space in ruins with no combat in case that balance consideration is important. You can steal the regular sleeper board from charlie so it's probably whatever

Mothblocks reviewed (changes requested):
rename it to `sleepers.dm` or something. this has absolutely fucked me over on my windows machine several times before and it drives me insane

JohnFulpWillard committed:
rename, and removal of syndie sleepers

Removes syndicate sleepers from non-syndicate maps. Now they only exist in the infiltrator, battlecruiser, and syndicate lavaland base.

JohnFulpWillard committed:
small mistake on snowdin

san7890 commented on code/game/machinery/sleepers.dm:
```diff
@@ -286,24 +321,41 @@
 	icon_state = "partypod"
 	base_icon_state = "partypod"
 	circuit = /obj/item/circuitboard/machine/sleeper/party
-	var/leddit = FALSE //Get it like reddit and lead alright fine
```
😢 i'll miss u leddit

Sealed101 commented:
should probably mention the sleeper swaps on the ruin in cl

JohnFulpWillard committed:
re-replaces sleepers with proper ones

JohnFulpWillard commented:
> should probably mention the sleeper swaps on the ruin in cl

Their functionality hasn't changed, they just look like NT ones instead of syndicate ones, if it's necessary I can add it but I feel like space ruins don't call for a changelog entry talking about how one item swapped colors

newfren commented:
syndicate sleepers have different functionality to nt ones in that syndicate ones can be used by the person inside while nt ones cannot

JohnFulpWillard commented:
> syndicate sleepers have different functionality to nt ones in that syndicate ones can be used by the person inside while nt ones cannot

I replaced them with ``/self_control`` subtype which does allow control from the inside

JohnFulpWillard commented on code/game/machinery/sleepers.dm:
```diff
@@ -161,21 +175,27 @@
 
 /obj/machinery/sleeper/process()
 	..()
-	check_nap_violations()
```
It checks it in ``ui_act``, i thought it was weird to have it in process too, I can re-add it if you want me to.

JohnFulpWillard commented on code/game/machinery/sleepers.dm:
```diff
@@ -161,21 +175,27 @@
+	use_power(idle_power_usage)
 
 /obj/machinery/sleeper/nap_violation(mob/violator)
+	. = ..()
```
No, but it was added in case functionality is ever added (i hope not)

JohnFulpWillard committed:
review comments

JohnFulpWillard committed:
i mixed it up with chem_buttons

Addust commented:
you should be able to get sleeper board from ruin sleepers but make the pried out boards have half max chemicals if you really think removing sleepers was a good idea

JohnFulpWillard commented:
> you should be able to get sleeper board from ruin sleepers but make the pried out boards have half max chemicals if you really think removing sleepers was a good idea

sleepers don't have max chemicals

Addust commented:
i mean the OD threshold limit, where if you spamclick a button it'll be greyed out, half the amount that the sleeper locks out the button on

JohnFulpWillard commented:
That is still infinite omizine, and every other chem in there. I like them as a small reward for taking over syndicate areas, getting it from anywhere in space is really lame.

SuperNovaa41 commented on code/game/machinery/sleepers.dm:
```diff
@@ -15,25 +9,52 @@
+	///Whether this sleeper can be deconstructed and drop the board, if its on mapload.
+	var/deconstructable = FALSE
+	///Message sent when a user enters the machine.
+	var/enter_message = "<span class='notice'><b>You feel cool air surround you. You go numb as your senses turn inward.</b></span>"
```
```suggestion
	var/enter_message = span_boldnotice("You feel cool air surround you. You go numb as your senses turn inward.")
```

SuperNovaa41 commented on code/game/machinery/sleepers.dm:
```diff
@@ -98,35 +118,33 @@
-	. = TRUE
-	if(..())
-		return
+	. = ..()
```
you removed the `if (.)` check why

JohnFulpWillard replied:
Parent is just a return, there was no point in having it.

SuperNovaa41 commented on code/game/machinery/sleepers.dm:
```diff
@@ -286,24 +325,41 @@
-
 	controls_inside = TRUE
+	deconstructable = TRUE
+	enter_message = "<span class='notice'><b>You're surrounded by some funky music inside the chamber. You zone out as you feel waves of krunk vibe within you.</b></span>"
```
```suggestion
	enter_message = span_boldnotice("You're surrounded by some funky music inside the chamber. You zone out as you feel waves of krunk vibe within you.")
```

SuperNovaa41 reviewed (changes requested):
looks fine other than these small things

JohnFulpWillard committed:
Update code/game/machinery/sleepers.dm

Co-authored-by: Seth Scherer <supernovaa41@gmx.com>

JohnFulpWillard committed:
Update code/game/machinery/sleepers.dm

Co-authored-by: Seth Scherer <supernovaa41@gmx.com>
